notebooks/data analysis/Pathfinder_system_other_columns.ipynb

```diff
@@ -0,0 +1,3769 @@
+     },
+     "execution_count": 7,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "resources.head()"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 38,
+   "outputs": [],
+   "source": [
+    "def check_diff_values(df, values):\n",
+    "    return (\n",
+    "        len(\n",
+    "            [\n",
+    "                _\n",
+    "                for i, row in df.iterrows()\n",
+    "                for col in df.columns\n",
+    "                for val in values\n",
+    "                if type(row[col]) is dict\n",
+    "                and (len(row[col].keys()) != len(values) or val not in row[col].keys())\n",
+    "            ]\n",
+    "        )\n",
+    "        > 0\n",
+    "    )"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 41,
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Klucze inne niż: max, value?\n",
+      "False\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(\"Klucze inne niż: max, value?\")\n",
+    "print(check_diff_values(pd.DataFrame(data=resources.focus), [\"max\", \"value\"]))"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 56,
+   "outputs": [],
+   "source": [
+    "max_val = {}\n",
+    "\n",
+    "for i, row in resources.iterrows():\n",
+    "    if type(row[\"focus\"]) is dict:\n",
+    "        max_val[(row[\"focus\"].get(\"max\"), row[\"focus\"].get(\"value\"))] = None"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 57,
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{(1, 1): None, (0, 0): None, (3, 3): None, (2, 2): None}\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(max_val)"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "### Resources\n",
+    "Zatem w bestiariuszach focus value == focus max\n",
+    "\n",
+    "Zakres wartości: 0 - 3\n",
+    "\n",
+    "Nie każdy potworek ma\n",
+    "\n",
+    "Można zredukować złożoneość tej kolumny - ma 1 \"pod kolumne\" z 2 wartościami, które są takie same dla danego potworka o\n",
```

Comment:
Nie mam pojęcia co to jest.

Reply:
Okej, to jest chyba ilość umiejętności które ma wzmocnione. I te umiejętności chyba mogą być bardzo różne.
Można uwzględnić czemu nie.